src/network/netdev/hsr.c

```diff
@@ -0,0 +1,128 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+/* Make sure the net/if.h header is included before any linux/ one */
+#include <net/if.h>
+#include <netinet/in.h>
+#include <linux/if_arp.h>
+
+#include "netlink-util.h"
+#include "networkd-manager.h"
+#include "string-table.h"
+#include "string-util.h"
+#include "strv.h"
+#include "hsr.h"
+
+static const char * const hsr_protocol_table[_NETDEV_HSR_PROTOCOL_MAX] = {
+        [NETDEV_HSR_PROTOCOL_HSR] = "hsr",
+        [NETDEV_HSR_PROTOCOL_PRP] = "prp",
+};
+
+DEFINE_STRING_TABLE_LOOKUP(hsr_protocol, HsrProtocol);
```

Comment:
Only from_string is used in this source.
Please use `DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING()`.